### PR TITLE
feat: password dialogs integration

### PR DESCRIPTION
**Pull request by jhasuraj01**
# Set Password:

Screenshot_20230916_135345

Screenshot_20230916_134739

# Change Password:

Screenshot_20230916_134921

Screenshot_20230916_144942

# Remove Password:

Screenshot_20230916_135156

Screenshot_20230916_135259

### CODE REVIEW

File: packages/cysync-core/src/dialogs/Password/ChangePassword/Dialogs/CreateNewPassword.tsx [hunk 1]
```diff
@@ -121,12 +125,14 @@ export const CreateNewPassword: React.FC = () => {
           type="submit"
           variant="primary"
           disabled={
+            loading ||
             Boolean(error) ||
             oldPassword.length === 0 ||
             newPassword.length < 8 ||
             confirmNewPassword.length < 8 ||
             newPassword !== confirmNewPassword
```
Comment (irshadCypherock) [inline packages/cysync-core/src/dialogs/Password/ChangePassword/Dialogs/CreateNewPassword.tsx]: All the validation checks should be in `context`

File: packages/cysync-core/src/dialogs/Password/ChangePassword/context/index.tsx [hunk 2]
```diff
@@ -57,15 +61,26 @@ export const ChangePasswordDialogProvider: FC<
   const [oldPassword, setOldPassword] = useState<string>('');
   const [newPassword, setNewPassword] = useState<string>('');
   const [confirmNewPassword, setConfirmNewPassword] = useState<string>('');
+  const [loading, setLoading] = useState<boolean>(false);
```
Comment (irshadCypherock) [inline packages/cysync-core/src/dialogs/Password/ChangePassword/context/index.tsx]:
```suggestion
  const [isLoading, setIsLoading] = useState<boolean>(false);
```

File: packages/cysync-core/src/dialogs/Password/RemovePassword/Dialogs/ConfirmPassword.tsx [hunk 3]
```diff
@@ -71,7 +78,8 @@ export const ConfirmPassword: React.FC = () => {
           form="remove-password-confirm-form"
           type="submit"
           variant="danger"
-          disabled={Boolean(error) || password.length === 0}
+          disabled={loading || Boolean(error) || password.length === 0}
```
Comment (irshadCypherock) [inline packages/cysync-core/src/dialogs/Password/RemovePassword/Dialogs/ConfirmPassword.tsx]: All validation should be in `context`

File: packages/cysync-core/src/dialogs/Password/RemovePassword/context/index.tsx [hunk 4]
```diff
@@ -51,15 +53,22 @@ export const RemovePasswordDialogProvider: FC<
 
   const [error, setError] = useState<string | null>(null);
   const [password, setPassword] = useState<string>('');
+  const [loading, setLoading] = useState<boolean>(false);
```
Comment (irshadCypherock) [inline packages/cysync-core/src/dialogs/Password/RemovePassword/context/index.tsx]: Same here

File: packages/cysync-core/src/dialogs/Password/SetPassword/Dialogs/AddPassword.tsx [hunk 5]
```diff
@@ -101,11 +104,13 @@ export const AddPassword: React.FC = () => {
           type="submit"
           variant="primary"
           disabled={
+            loading ||
             Boolean(error) ||
             newPassword.length < 8 ||
             confirmNewPassword.length < 8 ||
             newPassword !== confirmNewPassword
```
Comment (irshadCypherock) [inline packages/cysync-core/src/dialogs/Password/SetPassword/Dialogs/AddPassword.tsx]: Same here

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: packages/cysync-core/src/pages/MainApp/Settings/Tabs/AppSettings.tsx [hunk 6]
```diff
@@ -27,19 +29,21 @@ export const AppSettings: React.FC = () => {
         description={{ text: item.password.description }}
       >
         <SettingsButton
-          display="none"
+          display={isPasswordSet ? 'none' : undefined}
           onClick={() => dispatch(openSetPasswordDialog())}
           variant="primary"
         >
```
Comment (irshadCypherock) [inline packages/cysync-core/src/pages/MainApp/Settings/Tabs/AppSettings.tsx]:
```suggestion
        {!isPasswordSet && (<SettingsButton
          onClick={() => dispatch(openSetPasswordDialog())}
          variant="primary"
        >)}
```

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: packages/cysync-core/src/dialogs/Password/ChangePassword/Dialogs/CreateNewPassword.tsx (hunk 1 above)
Comment (jhasuraj01) [reply packages/cysync-core/src/dialogs/Password/ChangePassword/Dialogs/CreateNewPassword.tsx]: validation checks moved to context

File: packages/cysync-core/src/dialogs/Password/ChangePassword/context/index.tsx (hunk 2 above)
Comment (jhasuraj01) [reply packages/cysync-core/src/dialogs/Password/ChangePassword/context/index.tsx]: updated variable name

Comment (jhasuraj01) [reply packages/cysync-core/src/dialogs/Password/ChangePassword/context/index.tsx:64]: updated variable name

File: packages/cysync-core/src/dialogs/Password/RemovePassword/context/index.tsx (hunk 4 above)
Comment (jhasuraj01) [reply packages/cysync-core/src/dialogs/Password/RemovePassword/context/index.tsx]: updated variable name

File: packages/cysync-core/src/dialogs/Password/RemovePassword/Dialogs/ConfirmPassword.tsx (hunk 3 above)
Comment (jhasuraj01) [reply packages/cysync-core/src/dialogs/Password/RemovePassword/Dialogs/ConfirmPassword.tsx]: validation moved to context

File: packages/cysync-core/src/dialogs/Password/SetPassword/Dialogs/AddPassword.tsx (hunk 5 above)
Comment (jhasuraj01) [reply packages/cysync-core/src/dialogs/Password/SetPassword/Dialogs/AddPassword.tsx]: validation moved to context

File: packages/cysync-core/src/pages/MainApp/Settings/Tabs/AppSettings.tsx (hunk 6 above)
Comment (jhasuraj01) [reply packages/cysync-core/src/pages/MainApp/Settings/Tabs/AppSettings.tsx]: Changes Implemented

**Review by irshadCypherock — APPROVED** [review]

**Pull request merged**